app/backend/wells/management/commands/export.py

```diff
@@ -0,0 +1,221 @@
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+
+    def handle(self, *args, **options):
+        logger.info('starting export')
+        zip_filename = 'gwells.zip'
+        spreadsheet_filename = 'gwells.xlsx'
+        self.generate_files(zip_filename, spreadsheet_filename)
+        self.upload_files(zip_filename, spreadsheet_filename)
+        logger.info('cleaning up')
+        for filename in (zip_filename, spreadsheet_filename):
+            if os.path.exists(filename):
+                os.remove(filename)
+        logger.info('export complete')
+
+    def upload_files(self, zip_filename, spreadsheet_filename):
+        minioClient = Minio(get_env_variable('S3_HOST'),
+                            access_key=get_env_variable('S3_PUBLIC_ACCESS_KEY'),
+                            secret_key=get_env_variable('S3_PUBLIC_SECRET_KEY'),
+                            secure=True)
+        for filename in (zip_filename, spreadsheet_filename):
+            logger.info('uploading {}'.format(filename))
+            with open(filename, 'rb') as file_data:
+                file_stat = os.stat(filename)
+                # Do we need to remove the existing files 1st?
+                minioClient.put_object(get_env_variable('S3_WELL_EXPORT_BUCKET'),
+                                       filename,
+                                       file_data,
+                                       file_stat.st_size)
+
+    def export(self, workbook, gwells_zip, worksheet_name, cursor):
+        logger.info('exporting {}'.format(worksheet_name))
+        worksheet = workbook.add_worksheet(worksheet_name)
+        csv_file = '{}.csv'.format(worksheet_name)
+        if os.path.exists(csv_file):
+            os.remove(csv_file)
+        with open(csv_file, 'w') as csvfile:
+            csvwriter = csv.writer(csvfile, dialect='excel')
+
+            values = []
+            # Write the headings
+            for index, field in enumerate(cursor.description):
+                worksheet.write(0, index, '{}'.format(field.name))
+                values.append(field.name)
+            csvwriter.writerow(values)
+
+            # Write the values
+            row_index = 0
+            for row, record in enumerate(cursor.fetchall()):
+                values = []
+                num_values = 0
+                for col, value in enumerate(record):
+                    if value:
+                        num_values += 1
+                    values.append(value)
+                if num_values > 1:
+                    # We always have a well_tag_number, but if that's all we have, then just skip this record
+                    row_index += 1
+                    for col, value in enumerate(values):
+                        if value:
+                            worksheet.write(row_index, col, value)
+                    csvwriter.writerow(values)
+        gwells_zip.write(csv_file)
+        if os.path.exists(csv_file):
+            os.remove(csv_file)
```

Comment:
is this because gwells_zip.write creates a different file?

Reply:
Yep - once the csv is thrown into the zip file, we can get rid of it.